### PR TITLE
[FIX] [16.0] stock_move_line_expiration_date_required: Remove required from field definition

### PR DESCRIPTION
**Pull request by Shide**
Using the required on the field definition causes the database to set "not nullable" on expiry_date column on stock.move.line. 

Moved the required to the views.

MT-4224 @moduon @yajo @EmilioPascual @rafaelbn please review

### CODE REVIEW

File: stock_move_line_expiration_date_required/models/stock_move_line.py
Comment (yajo) [inline stock_move_line_expiration_date_required/models/stock_move_line.py:14]: You'll need a migration script to remove the constraint that was wrongly created by this.

**Review by yajo — CHANGES_REQUESTED** [review]

File: stock_move_line_expiration_date_required/models/stock_move_line.py
Comment (Shide) [reply stock_move_line_expiration_date_required/models/stock_move_line.py:14]: Done. I've re-launched the `update_db` method on the `expiration_date` field.

**Comment by Shide** [comment]
I think the ocabot merge will need to increment the verison of the module

File: stock_move_line_expiration_date_required/migrations/16.0.1.0.1/post-migration.py
Comment (rousseldenis) [inline stock_move_line_expiration_date_required/migrations/16.0.1.0.1/post-migration.py]: @Shide The current version of this module is 16.0.1.0.1 yet. So, you need both to change this directory name and manifest version

**Review by fcvalgar — APPROVED** [review]
Functional review.

LGTM, thank you @Shide.

https://www.loom.com/share/ec7071afa48644b69199d3d7bb8a537a?sid=22e04f0f-1975-4514-9126-db27f4847bac

File: stock_move_line_expiration_date_required/migrations/16.0.1.0.1/post-migration.py
Comment (Shide) [reply stock_move_line_expiration_date_required/migrations/16.0.1.0.1/post-migration.py]: Done, thanks! :)

**Review by rafaelbn — APPROVED** [review]
Functionaly 👍🏼 

I think we should wait for @yajo or @rousseldenis approval to merge, technical review, thanks

File: stock_move_line_expiration_date_required/__manifest__.py
Comment (yajo) [inline stock_move_line_expiration_date_required/__manifest__.py:6]: **suggestion(non-blocking):** usually, you shouldn't use migration scripts in patch versions. That should indicate that the version bump should be major.

According to [our guidelines](https://github.com/OCA/odoo-community.org/blob/8afdb5a5ebfd3314e2fc836c9b9f644468ee044e/website/Contribution/CONTRIBUTING.rst#L71-L72), it'd be a major or minor bump.

Future-proofing the code, [it's better to restrict migration scripts to major or minor versions too](https://github.com/orgs/OCA/discussions/117#discussion-5699815).

However, this one PR is a little bit crossing the line because it's really a bugfix. Thus, this is not required at this time IMHO. But at least there's some extra info for you to keep in mind for next times :)

**Review by yajo — APPROVED** [review]

**Pull request merged**
